packages/teams-js/test/public/pages.spec.ts

```diff
@@ -24,98 +29,216 @@ describe('AppSDK-TeamsAPIs', () => {
+        it(`pages.returnFocus should not allow calls from ${context} context`, async () => {
           await utils.initializeWithContext(context);
-
-          await expect(pages.navigateToApp(navigateToAppParams)).rejects.toThrowError(
-            `This call is only allowed in following contexts: ["content","sidePanel","settings","task","stage","meetingStage"]. Current context: "${context}".`,
+          expect(() => pages.returnFocus()).toThrowError(
+            `This call is only allowed in following contexts: ${JSON.stringify(
+              allowedContexts,
+            )}. Current context: "${context}".`,
           );
         });
       }
     });
+  });
 
-    it('should successfully send the navigateToApp message', async () => {
-      await utils.initializeWithContext('content');
-      utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
+  describe('Testing pages.registerFocusEnterHandler function', () => {
+    it('pages.registerFocusEnterHandler should not allow calls before initialization', () => {
+      expect(() => pages.registerFocusEnterHandler(emptyCallback)).toThrowError(
+        'The library has not yet been initialized',
+      );
+    });
+    Object.values(FrameContexts).forEach(context => {
+      it(`pages.registerFocusEnterHandler should successfully register a focus enter handler when initialized with ${context} context`, async () => {
+        await utils.initializeWithContext(context);
+        pages.registerFocusEnterHandler((x: boolean) => {
+          return true;
+        });
+        const messageForRegister = utils.findMessageByFunc('registerHandler');
+        expect(messageForRegister).not.toBeNull();
+        expect(messageForRegister.args.length).toBe(1);
+        expect(messageForRegister.args[0]).toBe('focusEnter');
+      });
+
+      it(`pages.registerFocusEnterHandler should successfully invoke focus enter handler when set to true and  initialized with ${context} context`, async () => {
+        await utils.initializeWithContext(context);
 
-      const promise = pages.navigateToApp(navigateToAppParams);
+        let handlerInvoked = false;
+        pages.registerFocusEnterHandler((x: boolean) => {
+          handlerInvoked = true;
+          return true;
+        });
+
+        utils.sendMessage('focusEnter');
+        expect(handlerInvoked).toBe(true);
+      });
 
-      const navigateToAppMessage = utils.findMessageByFunc('pages.navigateToApp');
-      utils.respondToMessage(navigateToAppMessage, true);
-      await promise;
+      it(`pages.registerFocusEnterHandler should not invoke focus enter handler when set to false initialized with ${context} context`, async () => {
+        await utils.initializeWithContext(context);
 
-      expect(navigateToAppMessage).not.toBeNull();
-      expect(navigateToAppMessage.args[0]).toStrictEqual(navigateToAppParams);
+        let handlerInvoked = true;
+        pages.registerFocusEnterHandler((x: boolean) => {
```

Comment:
I assume that these functions are supposed to return a boolean, because we aren't doing anything with it

Reply:
Yes, are just verifying that the handler is invoked or not by flowing through the code.